### PR TITLE
OK-7814 `ChangeLogModal` component and `Modal` maxHeight prop

### PR DESCRIPTION
**Pull request by franco-chan**
### ✨ How to use
``` js
const md = `
  # What's New in 1.0.3
  ### Heading3
  Paragraph..
  - Bulletpoint...
  - Bulletpoint...
  ---
  # What's new in 1.0.2
  ...
`;

<ChangeLogModal md={md} />
```

### 💎 What Changed
- Modified `Markdown` component so it can match our design system styles.
- Now we have `maxHeight` prop for `Modal` component, it only effects the max height of Modal on desktop.
- The `headerShown` prop was added to determine whether or not the `Modal.Header` should be displayed.

### CODE REVIEW

File: packages/components/src/Markdown/index.tsx [hunk 1]
```diff
@@ -1,23 +1,70 @@
...
 const Markdown: FC = ({ children }) => {
-  const textColor = useThemeValue('text-default');
-  const bgColor = useThemeValue('surface-default');
+  const headingColor = useThemeValue('text-default');
+  const bodyTextColor = useThemeValue('text-subdued');
+  // const bgColor = useThemeValue('surface-default');
+  const isSmallScreen = useIsVerticalLayout();
+  const horizontalRulesColor = useThemeValue('divider');
+
+  const styles = StyleSheet.create({
```
Comment (revan-zhang) [inline packages/components/src/Markdown/index.tsx]: 这个其实可以 useMemo 一下

File: packages/components/src/Markdown/index.tsx [hunk 2]
```diff
@@ -1,23 +1,70 @@
...
-      style={{
-        body: { color: textColor },
-        code_inline: { padding: 2, backgroundColor: bgColor, borderWidth: 0 },
-        blockquote: { backgroundColor: bgColor, borderWidth: 0 },
-        code_block: { backgroundColor: bgColor, borderWidth: 0 },
-        fence: { backgroundColor: bgColor, borderWidth: 0 },
-      }}
+      mergeStyle={false}
+      style={styles}
+      // style={{
```
Comment (revan-zhang) [inline packages/components/src/Markdown/index.tsx]: 是不是最好删除这种已经确定没用的代码

File: packages/components/src/Markdown/index.tsx (hunk 1 above)
Comment (franco-chan) [reply packages/components/src/Markdown/index.tsx]: done

File: packages/components/src/Markdown/index.tsx (hunk 2 above)
Comment (franco-chan) [reply packages/components/src/Markdown/index.tsx]: done

**Review by revan-zhang — DISMISSED** [review]

**Review by kwoktung — DISMISSED** [review]

**Review by sidmorizon — DISMISSED** [review]

**Review by revan-zhang — APPROVED** [review]

**Review by linleiqin — APPROVED** [review]

**Review by kwoktung — APPROVED** [review]

**Pull request merged**